tests/unit/test-firefox/test.remote.js

```diff
@@ -375,4 +376,38 @@ describe('firefox.remote', () => {
 
   });
 
+  describe('findFreeTcpPort', async () => {
```

Comment:
Let's summarize the changes needed on this test case and how its final version should look like:

- we don't want to call the function targeted by the test case from the `describe` callback
- we want to be sure that the tcp port we got is free and the user that executes web-ext can successfully allocate a TCP server on it
- if we call findFreeTcpPort one more time and the previously returned port is not free anymore, we expect to get a different free tcp port

The final version should look more or less like the following:

```js
  describe('findFreeTcpPort', () => {
    async function promiseServerOnPort(port): Promise<net.Server> {
      return new Promise((resolve) => {
        const srv = net.createServer();
        srv.listen(port, () => {
          resolve(srv);
        });
      });
    }

    it('resolves to a free tcp port', async () => {
      const port = await findFreeTcpPort();
      assert.isNumber(port);
      // Expect a port that is not in the reserved range.
      assert.isAtLeast(port, 1024);

      // The TCP port can be used to successfully start a TCP server.
      const srv = await promiseServerOnPort(port);
      assert.equal(srv.address().port, port);

      // Check that calling tcp port again doesn't return the
      // previous port (as it is not free anymore).
      const newPort = await findFreeTcpPort();
      assert.notStrictEqual(port, newPort);
      assert.isAtLeast(port, 1024);

      // The new TCP port can be used to successfully start a TCP server.
      const srv2 = await promiseServerOnPort(newPort);
      assert.equal(srv2.address().port, newPort);

      srv.close();
      srv2.close();
    });

  });
```

In this version we have defined in the `describe` callback a test helper function named `promiseServerOnPort` which return a promise that resolves to a TCP server listening on the given TCP port.

The single test case named `"resolves to a free tcp port"` is calling `findFreeTcpPort` twice, it asserts the port we got is like we expected it, then starts two tcp servers (and assert that the TCP server is successfully started exactly on the given TCP port) and asserts that the two resolved ports are different.

Reply:
Okay! Thanks!